search6/src/main/java/org/hibernate/performance/search/model/application/ModelServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+package org.hibernate.performance.search.model.application;
+
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.hibernate.Session;
+import org.hibernate.search.engine.search.predicate.SearchPredicate;
+import org.hibernate.search.engine.search.predicate.dsl.MatchPredicateOptionsStep;
+import org.hibernate.search.engine.search.projection.SearchProjection;
+import org.hibernate.search.engine.search.projection.dsl.FieldProjectionValueStep;
+import org.hibernate.search.mapper.orm.Search;
+import org.hibernate.search.mapper.orm.automaticindexing.AutomaticIndexingStrategyName;
+import org.hibernate.search.mapper.orm.automaticindexing.session.AutomaticIndexingSynchronizationStrategyNames;
+import org.hibernate.search.mapper.orm.cfg.HibernateOrmMapperSettings;
+import org.hibernate.search.mapper.orm.common.EntityReference;
+import org.hibernate.search.mapper.orm.schema.management.SchemaManagementStrategyName;
+import org.hibernate.search.mapper.orm.scope.SearchScope;
+import org.hibernate.search.mapper.orm.session.SearchSession;
+import org.hibernate.search.util.common.data.RangeBoundInclusion;
+
+public class ModelServiceImpl implements ModelService {
+
+	@Override
+	public Properties properties(Kind kind) {
+		Properties config = new Properties();
+		config.put( HibernateOrmMapperSettings.SCHEMA_MANAGEMENT_STRATEGY,
+				SchemaManagementStrategyName.DROP_AND_CREATE_AND_DROP );
+		config.put( HibernateOrmMapperSettings.MAPPING_CONFIGURER, new SearchProgrammaticMapping() );
+		config.put( HibernateOrmMapperSettings.AUTOMATIC_INDEXING_SYNCHRONIZATION_STRATEGY,
+				AutomaticIndexingSynchronizationStrategyNames.WRITE_SYNC );
+
+		if ( kind.isLucene() ) {
+			config.put( "hibernate.search.backend.directory.type", "local-heap" );
```

Comment:
Same as Search 5: this should be `filesystem`, and you should provide a root path for the indexes.